packages/indy-vdr/src/anoncreds/indyVdrAnoncredsRegistry.ts

```diff
@@ -0,0 +1,375 @@
+            credentialDefinitionMetadata: {
+              didIndyNamespace: pool.indyNamespace,
+            },
+            resolutionMetadata: {},
+          } 
+        }
+  
+        agentContext.config.logger.error(`Error retrieving credential definition '${credentialDefinitionId}'`)
+  
+        return {
+          credentialDefinitionId,
+          credentialDefinitionMetadata: {},
+          resolutionMetadata: {
+            error: 'notFound',
+            message: `unable to resolve credential definition with id ${credentialDefinitionId}`,
+          },
+        }
+  
+      } catch (error) {
+        agentContext.config.logger.error(`Error retrieving credential definition '${credentialDefinitionId}'`, {
+          error,
+          credentialDefinitionId,
+        })
+  
+        return {
+          credentialDefinitionId,
+          credentialDefinitionMetadata: {},
+          resolutionMetadata: {
+            error: 'notFound',
+            message: `unable to resolve credential definition: ${error.message}`,
+          },
+        }
+      }
+    }
+  
+    public async registerCredentialDefinition(agentContext: AgentContext, options: IndyVdrRegisterCredentialDefinitionOptions): Promise<RegisterCredentialDefinitionReturn> {
+       // Make sure didIndyNamespace is passed
+       if (!options.options.didIndyNamespace) {
+        return {
+          credentialDefinitionMetadata: {},
+          registrationMetadata: {},
+          credentialDefinitionState: {
+            reason: 'no didIndyNamespace defined in the options. didIndyNamespace is required when using the Indy SDK',
+            credentialDefinition: options.credentialDefinition,
+            state: 'failed',
+          },
+        }
+      }
+  
+      try {
+        const indyVdrPoolService = agentContext.dependencyManager.resolve(IndyVdrPoolService);
+  
+        //TODO: get schema SeqNo
+        const credentialDefinitionId = getLegacyCredentialDefinitionId(options.options.didIndyNamespace, options.credentialDefinition., options.credentialDefinition.tag),
+  
+        const credentialDefinitionRequest = new CredentialDefinitionRequest({
+          submitterDid: options.credentialDefinition.issuerId,
+          credentialDefinition: {
+              ver: '1.0',
+              id: credentialDefinitionId,
+              schemaId: options.credentialDefinition.schemaId,
+              type: 'CL',
+              tag: options.credentialDefinition.tag,
+              value: {
+                  primary: options.credentialDefinition.value,
+                  
+              },
+          },
+        })
+  
+        const pool = indyVdrPoolService.getPoolForNamespace(options.options.didIndyNamespace);
+  
+        const didsApi = agentContext.dependencyManager.resolve(DidsApi)
+        const didResult = await didsApi.resolve(`did:sov:${options.credentialDefinition.issuerId}`)
+    
+        if (!didResult.didDocument) throw new AriesFrameworkError('Unable to resolve did')
```

Comment:
Instead of throwing an error, we can directly return the error message:

```suggestion
        if (!didResult.didDocument) {
        
        return {
          credentialDefinitionMetadata: {},
          registrationMetadata: {},
          credentialDefinitionState: {
            credentialDefinition: options.credentialDefinition,
            state: 'failed',
            reason:`didNotFound: unable to resolve did did:sov${options.credentialDefinition.issuerId}: ${didResult.didResolutionMetadata.message}`
          }
        }
        }
```